### ISSUE TITLE
[return-await] await should be required in catch block only when it has a finally block

### ISSUE THREAD

**Issue by danthegoodman** [typescript-eslint/typescript-eslint#2344]
**Repro**

```ts
async function main(){
  try {
    throw "foo"
  } catch(e) {
    return Promise.resolve("bar");
  }
}
```

```JSON
{
  "rules": {
    "no-return-await": ["error"],
    "@typescript-eslint/return-await": ["error", "in-try-catch"]
  }
}
```

**Expected Result**

No error

**Actual Result**

Returning an awaited promise is required in this context. (line 5)

**Additional Info**

This issue can cause this rule to be in conflict with eslint's `no-return-await`. Adding the await to resolve `@typescript-eslint/return-await` will then trigger `no-return-await`

> `function hasErrorHandler(node)`: Determines whether a thrown error from this node will be caught/handled within this function rather than immediately halting this function. For example, a statement in a `try` block will always have an error handler. A statement in a `catch` block will only have an error handler if there is also a `finally` block.
> <https://github.com/eslint/eslint/blob/master/lib/rules/no-return-await.js#L51-L58>

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.1` |
| `@typescript-eslint/parser`        | `3.7.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `12.16.1` |
| `yarn`                              | `1.22.4` |

**Comment by bradzacher** [comment]
> This issue can cause this rule to be in conflict with eslint's no-return-await.

You shouldn't be using the two rules in tandem. This rule is intended to be a better version of the base rule.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md#how-to-use

**Comment by danthegoodman** [comment]
@bradzacher Ah, good to know! Thank you for pointing that out.

**Comment by soobing** [comment]
I will try this Issue 😊
